Mathlib/LinearAlgebra/SymmetricAlgebra/Basic.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2025 Raphael Douglas Giles. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Raphael Douglas Giles, Zhixuan Dai, Zhenyan Fu, Yiming Fu, Jingting Wang
+-/
+import Mathlib.LinearAlgebra.TensorAlgebra.Basic
+
+/-!
+# Symmetric Algebras
+
+Given a commutative semiring `R`, and an `R`-module `M`, we construct the symmetric algebra of `M`.
+This is the free commutative `R`-algebra generated (`R`-linearly) by the module `M`.
+
+## Notation
+
+1. `SymmetricAlgebra R M` is a concrete construction of the symmetric algebra defined as a
+   quotient of the tensor algebra. It is endowed with an R-algebra structure and a commutative
+   ring structure.
+2. `SymmetricAlgebra.ι R` is the canonical R-linear map `M →ₗ[R] SymmetricAlgebra R M`.
+3. Given a morphism `ι : M →ₗ[R] A`, `IsSymmetricAlgebra ι` is a proposition saying that the algebra
+   homomorphism from `SymmetricAlgebra R M` to `A` lifted from `ι` is bijective.
+4. Given a linear map `f : M →ₗ[R] A'` to an commutative R-algebra `A'`, and a morphism
+   `ι : M →ₗ[R] A` with `p : IsSymmetricAlgebra ι`, `IsSymmetricAlgebra.lift p f`
+   is the lift of `f` to an `R`-algebra morphism `A →ₐ[R] A'`.
+
+-/
+
+universe u
+
+variable (R M : Type*) [CommSemiring R] [AddCommMonoid M] [Module R M]
+
+open TensorAlgebra in
+/-- Relation on the tensor algebra which will yield the symmetric algebra when
+quotiented out by. -/
+inductive SymRel : TensorAlgebra R M → TensorAlgebra R M → Prop where
```

Comment:
```suggestion
open TensorAlgebra in
/-- Relation on the tensor algebra which will yield the symmetric algebra when
quotiented out by. -/
inductive TensorAlgebra.SymRel : TensorAlgebra R M → TensorAlgebra R M → Prop where
```
to avoid this going in the global namespace.